sysboundary/ionosphere.h

```diff
@@ -148,20 +151,130 @@ namespace SBC {
          Rees1963, // Rees (1963)
          Rees1989, // Rees (1989)
          SergienkoIvanov, // Sergienko & Ivanov (1993)
+         Fang
       } ionizationModel;
       
+      bool precipitatingProtons; // If true and we use the Fang ionization models, include the proton component.
+      
+      enum Particles {
+         PROTON,
+         ELECTRON
+      };
+      
+      /*!< Lines  0-11 Fang 2013 proton Pij coefficients https://doi.org/10.1002/jgra.50484
+       *   Lines 12-19 Fang 2010 electron Pij coefficients https://doi.org/10.1029/2010GL045406
+       *   Lines 20-23 Padding with zeros to simplify calls.
+       */
+      constexpr static std::array<std::array<Real, 4>, 24> P_ij = {{ // hooray for C++ initialization, the additional {} is needed.
+         // proton
+         { 2.55050E+0,   2.69476E-1,  -2.58425E-1,   4.43190E-2},
+         { 6.39287E-1,  -1.85817E-1,  -3.15636E-2,   1.01370E-2},
+         { 1.63996E+0,   2.43580E-1,   4.29873E-2,   3.77803E-2},
+         {-2.13479E-1,   1.42464E-1,   1.55840E-2,   1.97407E-3},
+         {-1.65764E-1,   3.39654E-1,  -9.87971E-3,   4.02411E-3},
+         {-3.59358E-2,   2.50330E-2,  -3.29365E-2,   5.08057E-3},
+         {-6.26528E-1,   1.46865E+0,   2.51853E-1,  -4.57132E-2},
+         { 1.01384E+0,   5.94301E-2,  -3.27839E-2,   3.42688E-3},
+         {-1.29454E-6,  -1.43623E-1,   2.82583E-1,   8.29809E-2},
+         {-1.18622E-1,   1.79191E-1,   6.49171E-2,  -3.99715E-3},
+         { 2.94890E+0,  -5.75821E-1,   2.48563E-2,   8.31078E-2},
+         {-1.89515E-1,   3.53452E-2,   7.77964E-2,  -4.06034E-3},
+         // electron
+         { 1.24616E+0,   1.45903E+0,  -2.42269E-1,   5.95459E-2},
+         { 2.23976E+0,  -4.22918E-7,   1.36458E-2,   2.53332E-3},
+         { 1.41754E+0,   1.44597E-1,   1.70433E-2,   6.39717E-4},
+         { 2.48775E-1,  -1.50890E-1,   6.30894E-9,   1.23707E-3},
+         {-4.65119E-1,  -1.05081E-1,  -8.95701E-2,   1.22450E-2},
+         { 3.86019E-1,   1.75430E-3,  -7.42960E-4,   4.60881E-4},
+         {-6.45454E-1,   8.49555E-4,  -4.28581E-2,  -2.99302E-3},
+         { 9.48930E-1,   1.97385E-1,  -2.50660E-3,  -2.06938E-3},
+         {          0,            0,            0,            0},
+         {          0,            0,            0,            0},
+         {          0,            0,            0,            0},
+         {          0,            0,            0,            0}
+      }};
+      
```

Comment:
Now this is really ugly.
Why does this need to be a single array, in which electrons just *happen* to start at index 13?
Can't this be a 2-dimensional array, with the just previously defined enum as an index? Or two separate ones altogether?

Reply:
Also, I suggest this be called fang_P_ij

Reply:
Uh, sure, this has evolved organically while I was writing out the papers in code form.